### DISCUSSION TITLE
引入mars3d的问题

### DISCUSSION THREAD

**Discussion by Eva-Skye** [zouyaoji/vue-cesium#248] [General]
![image](https://user-images.githubusercontent.com/45868259/184051329-8f556b89-9b07-4689-8897-8f4b03053446.png)
例子写的不够详细，官方写的示例和描述不符，libs怎么配置能不能给个示例呢，另外mars3d能不能通过npm install 的模块引入

**Comment by Eva-Skye** [comment]
补充一下，我是使用vue-cesium-vite-starter原工程，只修改了main.ts，在use VueCesium的时候，将cesiumPath去掉，加上了示例的mars3dConfig，地球就没有正确加载，控制台中也没有打印出火星sdk不让去除的log，map是undefined

**Comment by Eva-Skye** [comment]
版本从3.0.12升到3.0.16可以加载了，但是能不能使用node modules里面的资源呢

**Reply by zouyaoji** [reply]
嗯，最新版本废弃了之前的mars3d的引入方式，具体可以参看 https://github.com/marsgis/mars3d/discussions/16 和 [更新日志 ](https://github.com/zouyaoji/vue-cesium/blob/dev/CHANGELOG.zh-CN.md#3016)。文档确实还有一块写的是 cesiumPath 没改过来，不好意思了，马上就去改。

可以使用node modules里面的资源，要麻烦点。根据你用的vite还是webpack，找对应的插件把你要加载的内容copy到public目录。假设复制到根目录，应该就是就写成下面那样：
```ts
const app = createApp(App)
app.use(VueCesium, {
  mars3dConfig: {
    include: 'mars3d',
    libs: {
            mars3d: [
            //三维地球“主库”
            '/mars3d-cesium/Build/Cesium/Widgets/widgets.css',
            '/mars3d-cesium/Build/Cesium/Cesium.js',
            '/mars3d/dist/mars3d.css',
            '/mars3d/dist/mars3d.js'
        ]
    }
  }
} as ConfigProviderContext)
```

其他的扩展库也是类似的，全部的情况请参考

https://github.com/zouyaoji/vue-cesium/blob/dev/packages/components/viewer/src/loadUtil.ts#L17

**Reply by Eva-Skye** [reply]
谢谢作者大大的及时回复，我刚开始使用vite，我看看能不能像webpack那样copy到public中去
